### DISCUSSION TITLE
can not find tools/make/targets.mk

### DISCUSSION THREAD

**Discussion by godhj93** [bitcraze/discussions#522] [Q&A]
Hi all,

I am trying to follow Session 3 of the Ai-deck-workshop.
(https://www.youtube.com/watch?v=o9asYPHxEB4, the slide can be found at https://github.com/pulp-platform/AI-deck-workshop)

I succeed the hello world example but failed to the following example that communicating between Crazyflie and AI deck (Slide 16)

I moved the example directories "bitcraze_cf_app" and "bitcraze_gap8" in same level "crazyflie-firmware" as follows:
```
|--- app_hello_world
|--- bitcraze_cf_app
|--- bitcraze_gap8
|--- crazyfile-firmware
```
and when I run the command "make" in "bitcraze_cf_app", it cannot find "targets.mk" in "crazefile-firmware" saying
```
../crazyflie-firmware/Makefile:133: tools/make/targets.mk: No such file or directory
make: *** No rule to make target 'tools/make/targets.mk'.  Stop.
```

The Makefile which is in "bitcraze_cf_app" is written as follows:
```
# enable app support
APP=1
APP_STACKSIZE=300

VPATH += src/
PROJ_OBJ += main.o
PROJ_OBJ += uart_dma_pulp.o
INCLUDES += -Iinc

CRAZYFLIE_BASE := ../crazyflie-firmware
include $(CRAZYFLIE_BASE)/Makefile

```

and the file targets.mk is in proper path 

```
/home/hj/Projects/landing_rl/crazyflie-firmware/tools/make/targets.mk
```

So I am very confused because the file is in right way but the system can not find it.

Do you guys have a solution?

**Comment by krichardsson** [comment]
Hi!

The build system has changed a bit since the workshop you are following was created. 
You can read more about [apps](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/app_layer/) and [out of tree builds](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/oot/) in our documentation to get some more insights.

Your make file needs to be updated and the easiest way to get it right is probably to look at one of out example apps, for instance [hello world](https://github.com/bitcraze/crazyflie-firmware/tree/master/examples/app_hello_world)
